docs/architecture/adr-064-batch-verification.md

```diff
@@ -0,0 +1,85 @@
+# ADR 064: Batch Verification
+
+## Changelog
+
+- January 28, 2021: Created (@marbar3778)
+
+## Context
+
+Tendermint uses public private key cryptography for validator signing. When a block is proposed and voted on validators sign a message representing acceptance or rejection of a block. These signatures are also used to verify previous blocks are correct if a node is syncing. Currently, Tendermint requires each signature to be verified individually, this leads to a slow down of block times.
```

Comment:
```suggestion
Tendermint uses public private key cryptography for validator signing. When a block is proposed and voted on, validators sign a message representing acceptance or rejection of a block. These signatures are also used to verify previous blocks are correct if a node is syncing. Currently, Tendermint requires each signature to be verified individually, this leads to a slow down of block times.
```

> When a block is proposed and voted on, validators sign a message representing acceptance or rejection of a block...

Do validators sign to signal rejection? I thought they prevote/precommit nil in such cases?